src/Orchard.Web/Modules/Orchard.MediaProcessing/Filters/MediaProcessingHtmlFilter.cs

```diff
@@ -0,0 +1,198 @@
+                Settings.Quality,
+                Settings.DensityThreshold);
+
+            return _profileManager.GetImageProfileUrl(src, profileName, null, filters.ToArray());
+        }
+
+        private FilterRecord CreateResizeFilter(int width, int height) {
+            // Because the images can be resized in the HTML editor, we must assume that the image is of the exact desired
+            // dimensions and that stretch is an appropriate mode. Note that the default is to never upscale images.
+            var state = new Dictionary<string, string> {
+                { "Width", width.ToString() },
+                { "Height", height.ToString() },
+                { "Mode", "Stretch" },
+                { "Alignment", "MiddleCenter" },
+                { "PadColor", "" }
+            };
+
+            return new FilterRecord {
+                Category = "Transform",
+                Type = "Resize",
+                State = FormParametersHelper.ToString(state)
+            };
+        }
+
+        private FilterRecord CreateFormatFilter(int quality, string format) {
+            var state = new Dictionary<string, string> {
+                { "Quality", quality.ToString() },
+                { "Format", format },
+            };
+
+            return new FilterRecord {
+                Category = "Transform",
+                Type = "Format",
+                State = FormParametersHelper.ToString(state)
+            };
+        }
+
+        private string ProcessImageAltContent(string imgTag) {
+            var src = GetAttributeValue(imgTag, "src");
+            var alt = GetAttributeValue(imgTag, "alt");
+
+            if (string.IsNullOrEmpty(alt) && !string.IsNullOrEmpty(src)) {
+                var text = Path.GetFileNameWithoutExtension(src).Replace("-", " ").Replace("_", " ");
+                imgTag = SetAttributeValue(imgTag, "alt", text);
+            }
+
+            return imgTag;
+        }
+
+        private string GetAttributeValue(string tag, string attributeName) {
+            var match = Regex.Match(tag, GetAttributeRegex(attributeName));
+
+            return match.Success ? match.Groups[1].Value : null;
+        }
+
+        private int GetAttributeValueInt(string tag, string attributeName) {
+            var value = GetAttributeValue(tag, attributeName);
+
+            return int.TryParse(value, out int result) ? result : 0;
+        }
+
+        private string SetAttributeValue(string tag, string attributeName, string value) {
+            var attributeRegex = GetAttributeRegex(attributeName);
+
+            if (Regex.IsMatch(tag, attributeRegex)) {
+                return Regex.Replace(tag, attributeRegex, $"{attributeName}=\"{value}\"");
+            }
+            else {
+                return tag.Insert(tag.Length - 1, $" {attributeName}=\"{value}\"");
+            }
+        }
+
+        private string GetAttributeRegex(string attributeName) => $@"\b{attributeName}\s*=\s*[""']?([^""'\s>]*)[""']?";
```

Comment:
Should be cached in a static concurrent dictionary and contain compiled regexes.